### ISSUE TITLE
Visual Studio Code Crashed reason 'oom', code: -536870904

### ISSUE THREAD

**Issue by DaveGetGit** [microsoft/vscode#206895]
Type: <b>Bug</b>

Run playwright e2e tests
FYI: I am only consuming 46% of the memory and 25% of the cpu.

![image](https://github.com/microsoft/vscode/assets/81608999/0bc69ebe-9155-4184-945c-fc86495a75a6)

Version: 1.87.0 (user setup)
Commit: 019f4d1419fbc8219a181fab7892ebccf7ee29a2
Date: 2024-02-27T23:41:44.469Z
Electron: 27.3.2
ElectronBuildId: 26836302
Chromium: 118.0.5993.159
Node.js: 18.17.1
V8: 11.8.172.18-electron.0
OS: Windows_NT x64 10.0.22631

VS Code version: Code 1.87.0 (019f4d1419fbc8219a181fab7892ebccf7ee29a2, 2024-02-27T23:41:44.469Z)
OS version: Windows_NT x64 10.0.22631

**Comment by RedCMD** [comment]
I suspect a single process is limited to 4GB
tho I have no idea

**Comment by oskarkorczakaz** [comment]
I have similar problem

**Comment by Greyh4t** [comment]
after updated to latest version, I have same issue

```
版本: 1.88.0 (user setup)
提交: 5c3e652f63e798a5ac2f31ffd0d863669328dc4c
日期: 2024-04-03T13:26:18.741Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Windows_NT x64 10.0.19044
```

**Comment by Greyh4t** [comment]
> after updated to latest version, I have same issue
> 
> ```
> 版本: 1.88.0 (user setup)
> 提交: 5c3e652f63e798a5ac2f31ffd0d863669328dc4c
> 日期: 2024-04-03T13:26:18.741Z
> Electron: 28.2.8
> ElectronBuildId: 27744544
> Chromium: 120.0.6099.291
> Node.js: 18.18.2
> V8: 12.0.267.19-electron.0
> OS: Windows_NT x64 10.0.19044
> ```

I found the reason, it was caused by internal private extension

**Comment by Plaichat-123** [comment]
> > 更新到最新版本后，我有同样的问题
> > ```
> > 版本: 1.88.0 (user setup)
> > 提交: 5c3e652f63e798a5ac2f31ffd0d863669328dc4c
> > 日期: 2024-04-03T13:26:18.741Z
> > Electron: 28.2.8
> > ElectronBuildId: 27744544
> > Chromium: 120.0.6099.291
> > Node.js: 18.18.2
> > V8: 12.0.267.19-electron.0
> > OS: Windows_NT x64 10.0.19044
> > ```
> 
> 我找到了原因，这是由内部私有扩展引起的
Hello, I also encountered this problem. How can I determine if my VSCode has private extensions installed?

**Comment by Greyh4t** [comment]
> encountered

Just disable the extensions you think have problems, especially the recently installed extensions, and then observe for a while to see if the memory usage of vscode is decreasing.

**Comment by rzhao271** [comment]
Hi all, is the issue still occurring for you on VS Code 1.96 or later?

If yes, does the issue occur when extensions are disabled? You can reload the window with extensions disabled by pressing Cmd+Shift+P to open the Command Palette and running the "Developer: Reload With Extensions Disabled" command.

If the issue does not occur when extensions are disabled, you can run [extension bisect](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect) to figure out which extension is causing the issue.

If the issue still occurs when extensions are disabled, could you attach a crash report following the steps at [Creating a Crash Report](https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-a-crash-report)?

**Comment by DaveGetGit** [comment]
> Hi all, is the issue still occurring for you on VS Code 1.96 or later?
> 
> If yes, does the issue occur when extensions are disabled? You can reload the window with extensions disabled by pressing Cmd+Shift+P to open the Command Palette and running the "Developer: Reload With Extensions Disabled" command.
> 
> If the issue does not occur when extensions are disabled, you can run [extension bisect](https://code.visualstudio.com/blogs/2021/02/16/extension-bisect) to figure out which extension is causing the issue.
> 
> If the issue still occurs when extensions are disabled, could you attach a crash report following the steps at [Creating a Crash Report](https://github.com/microsoft/vscode/wiki/Native-Crash-Issues#creating-a-crash-report)?


Hey @rzhao271 
It has been resolved starting from version 19.6 in my context.
Thanks
